File: temp/fix-slides4.js
const fs = require('fs');
const path = require('path');

const baseDir = 'D:/新课开发/供应链/AI版/11 文化基建：供应链自动化推进中不能松手的一线信任与安全感/PPT/slides';

const files = [
  'slide-29.js', 'slide-37.js', 'slide-40.js', 'slide-41.js',
  'slide-43.js', 'slide-44.js', 'slide-46.js', 'slide-54.js',
  'slide-56.js', 'slide-77.js', 'slide-79.js', 'slide-80.js',
  'slide-99.js', 'slide-100.js', 'slide-101.js', 'slide-105.js',
  'slide-106.js', 'slide-108.js'
];

function fixLine(line) {
  // Check if line has any quotes at all
  if (!line.includes('"')) return line;

  // Find all occurrences of " and track which are string delimiters vs inner quotes
  // A string delimiter is:
  // - The first " after a ( or : or = or similar (opening delimiter)
  // - The last " before a , or ) or similar (closing delimiter)

  // Count total quotes
  const quotePositions = [];
  for (let i = 0; i < line.length; i++) {
    if (line[i] === '"') quotePositions.push(i);
  }

  if (quotePositions.length <= 2) return line; // No inner quotes possible

  // For lines with more than 2 quotes, we need to identify inner quotes
  // Strategy: for each pair of quotes, determine if they're delimiters or content

  // Simple heuristic: quotes that are adjacent to Chinese characters are likely inner quotes
  // because proper string delimiters usually have space or punctuation before/after

  let result = '';
  let i = 0;
  let innerQuoteCount = 0;

  while (i < line.length) {
    if (line[i] !== '"') {
      result += line[i];
      i++;
      continue;
    }

    // We're at a quote
    const quotePos = i;

    // Find context: what comes before and after this quote
    let before = '';
    let after = '';
    let bi = quotePos - 1;
    while (bi >= 0 && line[bi] === ' ') bi--;
    if (bi >= 0) before = line[bi];

    let ai = quotePos + 1;
    while (ai < line.length && line[ai] === ' ') ai++;
    if (ai < line.length) after = line[ai];

    // Check if this quote is between two Chinese characters
    // If so, it's definitely an inner quote
    const isBeforeChinese = /[\u4e00-\u9fff]/.test(before);
    const isAfterChinese = /[\u4e00-\u9fff]/.test(after);

    if (isBeforeChinese && isAfterChinese) {
      // Quote between Chinese chars - inner quote
      result += innerQuoteCount % 2 === 0 ? '「' : '」';
      innerQuoteCount++;
      i++;
      continue;
    }

    // Also check if quote is right after a Chinese char and before a non-quote
    if (isBeforeChinese && after !== '"' && after !== ',' && after !== ')' && after !== ';' && after !== '{' && after !== '}') {
      // Quote after Chinese, not followed by typical closing chars
      result += '」';
      innerQuoteCount++;
      i++;
      continue;
    }

    // Check if quote is followed by Chinese char (opening inner quote)
    if (isAfterChinese && before !== '(' && before !== ':' && before !== '=' && before !== ' ' && before !== '"') {
      // Quote before Chinese, not preceded by typical opening chars
      result += '「';
      innerQuoteCount++;
      i++;
      continue;
    }

    // Otherwise, treat as regular quote (delimiter or actual content)
    result += line[i];
    i++;
  }

  return result;
}

let fixed = 0;
let verified = 0;
let errors = [];

for (const file of files) {
  const filePath = path.join(baseDir, file);

  try {
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');
    const fixedLines = [];

    let changed = false;
    for (const line of lines) {
      const fixedLine = fixLine(line);
      if (fixedLine !== line) changed = true;
      fixedLines.push(fixedLine);
    }

    const fixedContent = fixedLines.join('\n');

    if (changed) {
      fs.writeFileSync(filePath, fixedContent, 'utf8');
      fixed++;
      console.log(`Fixed: ${file}`);
    } else {
      console.log(`No changes: ${file}`);
    }

    // Verify by requiring the module
    try {
      require(filePath);
      verified++;
      console.log(`  Verified OK: ${file}`);
    } catch (e) {
      errors.push({ file, error: e.message });
      console.log(`  Verification FAILED: ${file} - ${e.message}`);
    }

  } catch (e) {
    console.log(`Error: ${file} - ${e.message}`);
    errors.push({ file, error: e.message });
  }
}

console.log('\n--- Summary ---');
console.log(`Fixed: ${fixed} files`);
console.log(`Verified OK: ${verified} files`);
console.log(`Errors: ${errors.length} files`);
if (errors.length > 0) {
  console.log('\nFailed files:');
  errors.forEach(e => console.log(`  ${e.file}: ${e.error}`));
}

File: temp/fix_slides4.py
import glob

def has_chinese(s):
    for c in s:
        o = ord(c)
        if 0x4e00 <= o <= 0x9fff:
            return True
    return False

def fix_content(content):
    result = []
    i = 0
    changed = False

    while i < len(content):
        idx = content.find('addText(', i)
        if idx == -1:
            result.append(content[i:])
            break

        result.append(content[i:idx])
        str_start = idx + len('addText(')

        if str_start >= len(content) or content[str_start] != '"':
            result.append(content[idx:str_start])
            i = str_start
            continue

        remaining = content[str_start+1:]
        quote_positions = []
        p = 0
        while p < len(remaining):
            if remaining[p] == '\\':
                p += 2
                continue
            if remaining[p] == '"':
                quote_positions.append(p)
                p += 1
            else:
                p += 1

        if not quote_positions:
            result.append(content[idx:str_start+1])
            i = str_start + 1
            continue

        end_quote_pos = None
        for qp in reversed(quote_positions):
            after = remaining[qp+1:qp+5].strip()
            if after.startswith(',') or after.startswith(')') or after == '':
                end_quote_pos = qp
                break

        if end_quote_pos is None:
            end_quote_pos = quote_positions[-1]

        inner = remaining[:end_quote_pos]

        if has_chinese(inner) and '"' in inner:
            fixed = []
            k = 0
            while k < len(inner):
                c = inner[k]
                if c == '"':
                    prev_c = inner[k-1] if k > 0 else ''
                    next_c = inner[k+1] if k < len(inner)-1 else ''

                    # Only replace if BOTH sides are Chinese
                    if (has_chinese(prev_c) and has_chinese(next_c)):
                        fixed.append('》')
                    else:
                        fixed.append(c)
                else:
                    fixed.append(c)
                k += 1

            new_inner = ''.join(fixed)
            if new_inner != inner:
                inner = new_inner
                changed = True

        result.append('addText("' + inner + '"')
        i = str_start + 1 + end_quote_pos + 1

    return ''.join(result), changed

fixed_count = 0
for f in sorted(glob.glob('slide-*.js')):
    with open(f, 'r', encoding='utf-8') as fh:
        content = fh.read()

    new_content, changed = fix_content(content)

    if changed:
        with open(f, 'w', encoding='utf-8') as fh:
            fh.write(new_content)
        print(f'Fixed: {f}')
        fixed_count += 1

print(f'Total fixed: {fixed_count}')
